core/src/main/scala/org/apache/spark/Partitioner.scala

```diff
@@ -138,7 +138,8 @@ class RangePartitioner[K : Ordering : ClassTag, V](
     partitions: Int,
     rdd: RDD[_ <: Product2[K, V]],
     private var ascending: Boolean = true,
-    val samplePointsPerPartitionHint: Int = 20)
+    val samplePointsPerPartitionHint: Int = 20,
+    needCacheSample: Boolean = false)
```

Comment:
can we create a different `RangePartitioner` in Spark SQL? it's a little weird to make a change in the core module which only make sense in the SQL module.